File: 백준/Silver/10828. 스택/스택.py
# 스택

import sys
input = sys.stdin.readline

st = []
# input N
N = int(input())


# loop
for _ in range(N):
    cmd = list(input().split())
    if cmd[0] == "push":
        st.append(int(cmd[1]))
        continue

    if cmd[0] == "pop":
        if len(st) == 0:
            print(-1)
        else:
            print(st.pop())
        continue

    if cmd[0] == "top":
        if len(st) == 0:
            print(-1)
        else:
            print(st[-1])
        continue

    if cmd[0] == "size":
        print(len(st))
        continue

    if cmd[0] == "empty":
        if len(st) == 0:
            print(1)
        else:
            print(0)
        continue

File: node.js/백준/Silver/10828. 스택/스택.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const n = Number(input[0]);
const result = [];

const outputs = [];

const POP = "pop";
const TOP = "top";
const SIZE = "size";
const EMPTY = "empty";

for (let i = 1; i <= n; i++) {
    const command = input[i].trim().split(" ");
    // console.log(command);
    if (command.length > 1) {
        result.push(parseInt(command[1]));
        continue;
    }

    if (command[0] === POP) {
        outputs.push(result.pop() ?? -1);
        continue;
    }

    if (command[0] === SIZE) {
        outputs.push(result.length);
    }

    if (command[0] === EMPTY) {
        const tmp = result.length > 0 ? 0 : 1;
        outputs.push(tmp);
    }

    if (command[0] === TOP) {
        outputs.push(result[result.length - 1] ?? -1);
    }
}

console.log(outputs.join("\n"));